pkg/bsql/user_syncer.go

```diff
@@ -136,6 +136,21 @@ func (s *userSyncer) CreateAccount(
 		return nil, nil, nil, err
 	}
 
+	previousAccountResource, err := s.validateAccount(ctx, provisioningConfig, queryInputs)
+	if err != nil {
+		if !errors.Is(err, ErrUnableFindResourceProvisioning) {
+			return nil, nil, nil, fmt.Errorf("failed to check existing account: %w", err)
+		}
+	}
+
+	if previousAccountResource != nil {
+		logger.Info("account resource is already created", zap.String("resource_id", previousAccountResource.GetId().GetResource()))
+
+		return &v2.CreateAccountResponse_SuccessResult{
+			Resource: previousAccountResource,
+		}, plaintextDataList, nil, nil
```

Comment:
Oh yeah, definitely remove the plaintextDataList from the return value, because those credentials are not the ones for the existing account.